migrations/sqls/20190310135013-inital-ddl-up.sql

```diff
@@ -0,0 +1,96 @@
+/* Replace with your SQL commands */
+/*dev_trap DDL */
+
+/*TraPO - Data defintion language */
+
+--USE dev_trapo;
+
+/*Create person.person table */
+CREATE TABLE IF NOT EXISTS person (
+  personid int NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  persontype nvarchar(32) NOT NULL, /*PersonType contraint at front end */
+  firstname nvarchar(255) NOT NULL,
+  lastname nvarchar(255) NOT NULL,
+  dob date NOT NULL,
+  gender nvarchar(32) NOT NULL,  /*Gender constraint at front end */
+  dateadded DATETIME DEFAULT CURRENT_TIMESTAMP,
+  datemodified DATETIME DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  addedby int NOT NULL /*Self Reference to Person ID*/,
+  modifiedby int NOT NULL /*Self Reference to Person ID*/,
+  FOREIGN KEY (addedby) REFERENCES person(personid),
+  FOREIGN KEY (modifiedby) REFERENCES person(personid)
+);
+
+
+/*Crate person Contanct table*/
+CREATE TABLE IF NOT EXISTS contact (
+  personid int NOT NULL, /*Foreign key reference to person table */
+  contactid int NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  primcontact varchar(32) NOT NULL,
+  primcarrier varchar(32) NOT NULL,
+  emercontact varchar(32) NULL,
+  email varchar(128) NULL,
+  dateadded DATETIME DEFAULT CURRENT_TIMESTAMP,
+  addedby int NOT NULL, /*Foreign key reference to person table */
+  FOREIGN KEY (personid) REFERENCES person(personid),
+  FOREIGN KEY (addedby) REFERENCES person(personid)
+);
+
+/*Create Demographics Table*/
+CREATE TABLE IF NOT EXISTS demographics (
+  personid int NOT NULL, /*foreing key reference to person table*/
+  demographicid int NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  origincountry varchar(32) NOT NULL,
+  languagehome varchar(32) NOT NULL,
+  engproficiency varchar(32) NOT NULL,
+  lenusresidence tinyint NULL, /*in years can be null of country of origin is US residence*/
```

Comment:
This took me a while to understand it meant "length of U.S. residence." Snake case and/or spelling out the whole word might make this easier to understand.

Would this be better to capture the date they came to the US? So that the number doesn't become out-of-date as time passes.